streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/core/graph/executiongraph/ExecutionEdge.java

```diff
@@ -0,0 +1,66 @@
+package org.ray.streaming.runtime.core.graph.executiongraph;
+
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+
+/**
+ * An edge that attaches two execution vertices.
+ */
+public class ExecutionEdge implements Serializable {
+
+  /**
+   * The source(upstream) execution vertex.
+   */
+  private final ExecutionVertex sourceVertex;
```

Comment:
This will form a circular reference between `ExecutionVertex ` and `ExecutionEdge `, which will be difficult to serialize to python process.

Reply:
Will only serialize vertex's id when serializing and get vertex object from graph by id when deserializing.